[DOCS] Add test python file and doc for validating multiple batches on doc

HaebichanGX opened the pull request:
Add all the files (python test file and the corresponding doc).




- [ ] Description of PR changes above includes a link to [an existing GitHub issue](https://github.com/great-expectations/great_expectations/issues)
- [ ] PR title is prefixed with one of: [BUGFIX], [FEATURE], [DOCS], [MAINTENANCE], [CONTRIB]
- [ ] Code is linted - run `invoke lint` (uses `black` + `ruff`)
- [ ] Appropriate tests and docs have been updated

For more details, see our [Contribution Checklist](https://docs.greatexpectations.io/docs/contributing/contributing_checklist), [Coding style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/code_style), and [Documentation style guide](https://docs.greatexpectations.io/docs/contributing/style_guides/docs_style).

After you submit your PR, keep the page open and **monitor the statuses of the various checks made by our continuous integration process at the bottom of the page. Please fix any issues that come up** and [reach out on Slack](https://greatexpectations.io/slack) if you need help. Thanks for contributing!

HaebichanGX committed:
Make first commit

Add all the files (python test file and the corresponding doc).

ghost commented:
<details open>
<summary><b>👇 <a href="https://app.codesee.io/r/reviews?pr=8291&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations">Click on the image</a> for a new way to code review</b>
</summary>

####

[![Review these changes using an interactive CodeSee Map](https://s3.us-east-2.amazonaws.com/maps.codesee.io/images/github/great-expectations/great_expectations/8291/e689b1cf/f84e5ddcb2a3dc48398ebc20b34cde63bd337f67.svg)](https://app.codesee.io/r/reviews?pr=8291&src=https%3A%2F%2Fgithub.com%2Fgreat-expectations%2Fgreat_expectations)

#### Legend
<img src="https://app.codesee.io/Legend_v3.png" alt="CodeSee Map legend" width="220" height="156">
</details>

HaebichanGX committed:
Update how_to_validate_multiple_batches_within_single_checkpoint.md

HaebichanGX committed:
make nit change

make nit change

HaebichanGX committed:
Add more reformatting changes

Add reformatting changes and add py test to test runner script

HaebichanGX committed:
Fix lint issue on test script runner

Remove empty trailing spaces on script runner python file

HaebichanGX committed:
Run black on how_to_validate_multiple_batches_within_single_checkpoint

Run black on how_to_validate_multiple_batches_within_single_checkpoint.py to reformat the code

HaebichanGX committed:
Fix test script runner

fix data location on test script runner

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

HaebichanGX committed:
Make lint and other changes

Make lint and other changes to conform to PR check.

HaebichanGX committed:
Merge branch 'add_validate_multiple_batches_doc' of https://github.com/great-expectations/great_expectations into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

anthonyburdi committed:
Merge branch 'develop' into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc

Rachel-Reverie committed:
- revision of the markdown guide.
- added some additional tags to the example script, to support the revised markdown guide.

Rachel-Reverie committed:
Merge remote-tracking branch 'origin/add_validate_multiple_batches_doc' into add_validate_multiple_batches_doc

Rachel-Reverie committed:
- added step to review validations as Data Docs.

kwcanuck commented on docs/docusaurus/docs/guides/validation/checkpoints/how_to_validate_multiple_batches_within_single_checkpoint.md:
```diff
@@ -0,0 +1,71 @@
+
+
+
+By default, a Checkpoint will only validate the last Batch included in a Batch Request. This guide will demonstrate how to validate multiple Batches identified by a single Batch Request using a Python loop and a Checkpoint's `validations` parameter. 
```
```suggestion
By default, a Checkpoint only validates the last Batch included in a Batch Request. Use the information provided here to learn how you can use a Python loop and the Checkpoint `validations` parameter to validate multiple Batches identified by a single Batch Request. 
```

kwcanuck commented on docs/docusaurus/docs/guides/validation/checkpoints/how_to_validate_multiple_batches_within_single_checkpoint.md:
```diff
@@ -0,0 +1,71 @@
+
+## Create a Batch Request with multiple Batches
+
+The following Python code will create a Batch Request that includes every available Batch in a Data Asset named `asset`:
```
```suggestion
The following Python code creates a Batch Request that includes every available Batch in a Data Asset named `asset`:
```

kwcanuck commented on docs/docusaurus/docs/guides/validation/checkpoints/how_to_validate_multiple_batches_within_single_checkpoint.md:
```diff
@@ -0,0 +1,71 @@
+
+When working with a Filesystem Data Source and organizing Batches, the `batching_regex` argument determines the inclusion of multiple Batches into a single Data Asset, with each file that matches the `batching_regex` resulting in a single Batch.
+
+On the other hand, when utilizing a SQL Datasource, Data Assets include a single Batch by default, which can then be split into multiple Batches using splitters.
```
```suggestion
SQL Datasource Data Assets include a single Batch by default. You can use splitters to split the single Batch into multiple Batches.
```

kwcanuck commented on docs/docusaurus/docs/guides/validation/checkpoints/how_to_validate_multiple_batches_within_single_checkpoint.md:
```diff
@@ -0,0 +1,71 @@
+
+## Convert the list of Batches into a list of Batch Requests
+
+Since a Checkpoint validates Batch Requests, but only validates the last Batch found in a Batch Request, the next step in this process is to convert the list of Batches into a list of Batch Requests that return the corresponding individual Batch.
```
```suggestion
A Checkpoint validates Batch Requests, but only validates the last Batch found in a Batch Request. You'll need to convert the list of Batches into a list of Batch Requests that return the corresponding individual Batch.
```

kwcanuck commented on docs/docusaurus/docs/guides/validation/checkpoints/how_to_validate_multiple_batches_within_single_checkpoint.md:
```diff
@@ -0,0 +1,71 @@
+
+## Build a validations list 
+
+A Checkpoint class's `validations` parameter consists of a list of dictionaries.  Each dictionary pairs one Batch Request with the Expectation Suite it should be validated against.  The following code creates a valid `validations` list where each Batch Request is associated with an Expectation Suite named `example_suite`.
```
```suggestion
A Checkpoint class's `validations` parameter consists of a list of dictionaries.  Each dictionary pairs one Batch Request with the Expectation Suite it should be validated against.  The following code creates a valid `validations` list and associates each Batch Request with an Expectation Suite named `example_suite`.
```

kwcanuck commented on docs/docusaurus/docs/guides/validation/checkpoints/how_to_validate_multiple_batches_within_single_checkpoint.md:
```diff
@@ -0,0 +1,71 @@
+
+## Run Checkpoint
+
+The `validations` list, containing the pairings of Batch Requests and Expectation Suites, can now be passed to a single Checkpoint instance which will validate each Batch Request against its corresponding Expectation Suite.  This effectively validates each Batch included in the original multiple-Batch Batch Request.
```
```suggestion
The `validations` list, containing the pairings of Batch Requests and Expectation Suites, can now be passed to a single Checkpoint instance that validates each Batch Request against its corresponding Expectation Suite. This effectively validates each Batch included in the original multiple-Batch Batch Request.
```

kwcanuck commented on docs/docusaurus/docs/guides/validation/checkpoints/how_to_validate_multiple_batches_within_single_checkpoint.md:
```diff
@@ -0,0 +1,71 @@
+
+## Review the Validation Results
+
+Once the validations have been run, the following code can be used to build and view the Validation Results as Data Docs:
```
```suggestion
After the validations run, use the following code to build and view the Validation Results as Data Docs.
```
**Note:** In some instances, you're introducing code samples with colons, and in others you're not. Pick a style and use it consistently.

kwcanuck reviewed (commented):
I hope you don't mind, but I did a quick review. Feel free to implement or ignore my suggestions.

HaebichanGX committed:
Update docs/docusaurus/docs/guides/validation/checkpoints/how_to_validate_multiple_batches_within_single_checkpoint.md

Co-authored-by: Rob Gray <104205257+kwcanuck@users.noreply.github.com>

HaebichanGX committed:
Update docs/docusaurus/docs/guides/validation/checkpoints/how_to_validate_multiple_batches_within_single_checkpoint.md

Co-authored-by: Rob Gray <104205257+kwcanuck@users.noreply.github.com>

HaebichanGX committed:
Update docs/docusaurus/docs/guides/validation/checkpoints/how_to_validate_multiple_batches_within_single_checkpoint.md

Co-authored-by: Rob Gray <104205257+kwcanuck@users.noreply.github.com>

HaebichanGX committed:
Update docs/docusaurus/docs/guides/validation/checkpoints/how_to_validate_multiple_batches_within_single_checkpoint.md

Co-authored-by: Rob Gray <104205257+kwcanuck@users.noreply.github.com>

HaebichanGX committed:
Update docs/docusaurus/docs/guides/validation/checkpoints/how_to_validate_multiple_batches_within_single_checkpoint.md

Co-authored-by: Rob Gray <104205257+kwcanuck@users.noreply.github.com>

HaebichanGX committed:
Merge branch 'develop' into add_validate_multiple_batches_doc

HaebichanGX committed:
Update docs/docusaurus/docs/guides/validation/checkpoints/how_to_validate_multiple_batches_within_single_checkpoint.md

Co-authored-by: Rob Gray <104205257+kwcanuck@users.noreply.github.com>

HaebichanGX committed:
Update docs/docusaurus/docs/guides/validation/checkpoints/how_to_validate_multiple_batches_within_single_checkpoint.md

Co-authored-by: Rob Gray <104205257+kwcanuck@users.noreply.github.com>

HaebichanGX commented:
@kwcanuck committed all those suggestions, thank you Rob!

HaebichanGX committed:
Update checkpoint

Update checkpoint by getting rid of simple checkpoint for just a regular checkpoint.

HaebichanGX committed:
Make lint and black changes

Make lint and black changes

HaebichanGX committed:
Merge branch 'develop' into add_validate_multiple_batches_doc

Rachel-Reverie reviewed (approved):
LGTM

HaebichanGX committed:
Merge branch 'develop' into add_validate_multiple_batches_doc

github-actions[bot] committed:
Merge develop into add_validate_multiple_batches_doc
